### ISSUE TITLE
Parsing zero padded floats

### ISSUE THREAD

**Issue by sfinkens** [pytroll/trollsift#27]
Parsing a string with zero padded floats

```python
>>> import trollsift
>>> trollsift.parse('{foo:04.1f}', '00.0')
```
fails with the following error:

<details>

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/site-packages/trollsift/parser.py", line 401, in parse
    keyvals = regex_formatter.extract_values(fmt, stri, full_match=full_match)
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/site-packages/trollsift/parser.py", line 324, in extract_values
    match = re.match(regex, stri)
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/re.py", line 175, in match
    return _compile(pattern, flags).match(string)
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/re.py", line 288, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/sre_compile.py", line 764, in compile
    p = sre_parse.parse(p, flags)
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/sre_parse.py", line 924, in parse
    p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, 0)
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/sre_parse.py", line 420, in _parse_sub
    not nested and not items))
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/sre_parse.py", line 810, in _parse
    p = _parse_sub(source, state, sub_verbose, nested + 1)
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/sre_parse.py", line 420, in _parse_sub
    not nested and not items))
  File "/cmsaf/cmsaf-ops3/sfinkens/conda/envs/girafe/lib/python3.7/sre_parse.py", line 648, in _parse
    source.tell() - here + len(this))
re.error: multiple repeat at position 48
```

</details>

The problematic regular expression is `^(?P<foo>[-+]?(\d+(\.\d*)?|\.\d+)([eE][-+]?\d+)?{4})$` . Composing on the other hand works fine:

```python
>>> trollsift.compose('{foo:04.1f}', {'foo': 0.0})
'00.0'
```

**Comment by sfinkens** [comment]
Same problem with 
```python
trollsift.parse('{foo:4.1f}', '12.3456')
```
So the zero padding doesn't seem to be the problem, but rather specifying the total number of digits.

**Comment by djhoese** [comment]
I think for parsing the regex code is pretty naive and simply says "they said 'f', replace with regex X". In this case that is:

https://github.com/pytroll/trollsift/blob/144b2be9490ecc65f6f6ac21f7f4f3fd28f29917/trollsift/parser.py#L136

So it might match more than anticipated but I don't think it should fail either. I know we use the number with `d` all the time so my guess is it is the `?` at the end of the `f` regex and then trollsift naively adds on "{4}". I think the wrong solution would be to wrap the whole `f` regex in parentheses so the `{4}` at the end would mean "there should be 4 float numbers". Obviously that is wrong though and instead we should either:

1. Remove the functionality that adds on the `{4}` part for `f` fields. That of course would match more than it should (any float), but I'm not sure we are smart about other format types either.
2. Add the ability to fill in these numbers (pre '.', post '.') in the regex. Maybe specify the regex in the dictionary linked to above as a tuple, then in the python code detect the tuple and then do the fancy work of saying "there should be this many before the period, this many digits after.
3. Do a naive float if only `f` is specified, otherwise do a complex float regex if the numbers are specified. Similar to 2 but maybe just implemented a little differently.

**Comment by mraspaud** [comment]
Closing, fixed by #28
